src/Umbraco.Core/Persistence/Repositories/Implement/ContentTypeRepository.cs

```diff
@@ -297,6 +298,10 @@ protected override void PersistUpdatedItem(IContentType entity)
 
         private void PersistHistoryCleanup(IContentType entity)
         {
+            // historyCleanup property is not mandatory for api endpoint if it's not present instantiate a default one.
+            // DocumentTypeSave doesn't handle this for us like ContentType constructors do.
+            entity.HistoryCleanup ??= new HistoryCleanup();
```

Comment:
We should not alter the existing entity, so lets replace this code with the changes I did in the `v8/bugfix/historycleanup-cleanup` branch:
 https://github.com/umbraco/Umbraco-CMS/blob/699403c5a715ed4e90f8d477ac835eda984c0277/src/Umbraco.Core/Persistence/Repositories/Implement/ContentTypeRepository.cs#L298-L310